ChargeRateValidator.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace BatteryManagementSystem
+{
+    public class ChargeRateValidator: IValidator<BatteryMonitor>
+    {
+        static readonly float maximum = 0.8f;
+
+        static readonly float minimum = 0.3f;
+
+        public bool IsValid(BatteryMonitor batteryMonitor)
+        {
+            return !(batteryMonitor.ChargeRate > maximum);
```

Comment:
Duplicate Code ....separate range check code....

batteryMonitor.ChargeRate > max

Reply:
Hi sir
I did not get this. can you give me some more hint.